test/unit_tests/braket/devices/test_local_simulator.py

```diff
@@ -506,19 +506,19 @@ def test_init_unregistered_backend():
 @pytest.mark.xfail(raises=NotImplementedError)
 def test_run_unsupported_type():
     sim = LocalSimulator(DummyCircuitSimulator())
-    sim.run("I'm unsupported")
+    sim.run("I'm unsupported").result()
```

Comment:
these tests are testing that the `run` method raises a NotImplementedError. By adding a `result` call, the test becomes more ambiguous and would pass if either raise the error (unwanted). If the run method is still raising the error, the result call shouldn't even be reached here

Reply:
Also, if you're updating tests that use the xfail decorator, I would recommend reformatting them to use `with pytest.raises(<ErrorType>, match=expected_str):`

Reply:
> these tests are testing that the `run` method raises a NotImplementedError. By adding a `result` call, the test becomes more ambiguous and would pass if either raise the error (unwanted). If the run method is still raising the error, the result call shouldn't even be reached here

Now run method returns a LocalQuantumTask. And only calling the result will make sure that the asynchronous task is completed and the error is raised. Since the task is running asynchronously in different thread we cannot catch it explicitly in the main thread unless we call the result or check the status

Reply:
Hm I'm still not sure, since in both cases the run method returns a LocalQuantumTask. The creation of the task doesn't have any blocking factors (queues/availability windows), so it seems like it could make sense to create the task synchronously (and raise any errors during task creation here), without needing to poll for results. I believe tasks created against the API fail quickly too; how come this is different?

Reply:
When .run() it returns a LocalQuantumTask and the running of the task happens on a different thread. If we want to catch any kind of errors synchronously in the main thread we will have to call .run().result()